src/core/mesh/qgstopologicalmesh.h

```diff
@@ -202,15 +291,38 @@ class CORE_EXPORT QgsTopologicalMesh
       bool allowUniqueSharedVertex,
       bool writeInVertices );
 
-    QgsMeshVertexCirculator vertexCirculator( int vertexIndex ) const;
+
+    //! Returns whether the two faces can be joined sharing the index \a commonIndex
     static bool facesCanBeJoinedWithCommonIndex( const QgsMeshFace &face1, const QgsMeshFace &face2, int commonIndex );
 
-    QSet<int> concernedFacesBy( const QList<int> faceIndex ) const;
+    //! Returns all faces indexes that are concerned by the face with index in \a faceIndex, that is sharing a least one vertex or one edge
+    QSet<int> concernedFacesBy( const QList<int> faceIndexes ) const;
+
+    // Following methods are used to retrieve free vertex without going through all the vertices container.
+    // For now, we use only QSet<int> with vertex indexes, but,
+    // maybe later we could use more sophisticated reference (spatial index?), to retrieve fre vertex in an extent
```

Comment:
move to cpp file this kind of implementation comments. Also best to add doxy-comments for both new functions